src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor(name) {
+    super(name);
 
-class Carnivore extends Animal {
-  // write your code here
+    this.hidden = false;
+  }
 }
 
+Herbivore.prototype.hide = function() {
+  this.hidden = true;
+};
+
+class Carnivore extends Animal {}
+
+Carnivore.prototype.bite = function(prey) {
+  if (!prey.hidden && !(prey instanceof Carnivore)) {
+    prey.health -= 50;
+  }
+
+  if (prey.health === 0) {
+    Animal.alive = Animal.alive.filter(animal => animal !== prey);
```

Comment:
Objects comparing is bad idea, let's instead check if animal has more than 0 health instead